core/src/main/java/org/elasticsearch/action/ingest/IngestActionFilter.java

```diff
@@ -110,8 +111,9 @@ void processBulkIndexRequest(Task task, BulkRequest original, String action, Act
                 logger.error("failed to execute pipeline for a bulk request", throwable);
                 listener.onFailure(throwable);
             } else {
+                long ingestTookInMillis = Math.max(1L, System.currentTimeMillis() - ingestStartTime);
```

Comment:
is 1L the proper lower bound? should it rather be 0? seems a side effect of comparing against 0 to check whether ingest has run, which I agree with Boaz about